src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -710,6 +710,49 @@ public static void TryStartNoGCRegion_SOHSize_LOHSize_BlockingCollection()
             }, options).Dispose();
         }
 
+        [Theory]
+        [OuterLoop]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public static void TryStartNoGCRegion_TotalSizeOutOfRange(long size)
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.TimeOut = TimeoutMilliseconds;
+            RemoteInvoke(sizeString =>
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("totalSize", () => GC.TryStartNoGCRegion(long.Parse(sizeString)));
+                return SuccessExitCode;
+            }, size.ToString(), options).Dispose();
+        }
+
+        [Theory]
+        [OuterLoop]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public static void TryStartNoGCRegion_LOHSizeInvalid(long size)
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.TimeOut = TimeoutMilliseconds;
+            RemoteInvoke(sizeString =>
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("lohSize", () => GC.TryStartNoGCRegion(1024, long.Parse(sizeString)));
+                return SuccessExitCode;
+            }, size.ToString(), options).Dispose();
+        }
+
+        [Fact]
+        [OuterLoop]
+        public static void TryStartNoGCRegion_TotalSizeLessThanLOHSize()
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.TimeOut = TimeoutMilliseconds;
+            RemoteInvoke(() =>
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("lohSize", () => GC.TryStartNoGCRegion(1024, 1152921504606846976));
```

Comment:
Could this just be [InlineData(1152921504606846976)] on the previous test rather than its own duplicated test?

Reply:
It could, I made it a new test because conceptually it's a different code path. I don't mind merging them, though.